app/src/main/java/cm/aptoide/pt/install/InstallManager.java

```diff
@@ -289,8 +289,10 @@ private boolean mapIndeterminate(Download download) {
     Install.InstallationStatus status = Install.InstallationStatus.UNINSTALLED;
     if (download != null) {
       switch (download.getOverallDownloadStatus()) {
-        case Download.FILE_MISSING:
         case Download.INVALID_STATUS:
+          status = Install.InstallationStatus.INITIAL_STATE;
+          break;
```

Comment:
Question: why install status -> initial state in case of a download invalid status?

Reply:
Invalid status is the status that is provided when the download was just initialized. We are mapping that to the install correct status which is INITIAL_STATE.